python/packages/autogen-ext/tests/memory/test_redis_memory.py

```diff
@@ -0,0 +1,258 @@
+from collections.abc import AsyncGenerator
+
+import pytest
+import pytest_asyncio
+from autogen_core.memory import MemoryContent, MemoryMimeType
+from autogen_core.model_context import BufferedChatCompletionContext
+from autogen_core.models import UserMessage
+from autogen_ext.memory.redis import RedisMemory, RedisMemoryConfig
+from pydantic import ValidationError
+from redis import ConnectionError, Redis
+from redisvl.exceptions import RedisSearchError
+
+# Skip tests if Redis instance is not avalable locally
+try:
+    # Attempt to connect to Redis by initializing and pinging
+    client = Redis.from_url("redis://localhost:6379")  # type: ignore[reportUnkownMemberType]
```

Comment:
Can we have two sets of tests: one uses mock and the other uses actual redis server. Skipping the second set of tests if the redis server is not available. See example for using Redis mock in https://github.com/microsoft/autogen/blob/main/python/packages/autogen-ext/tests/cache_store/test_redis_store.py

For GitHub CI tests, add a separate workflow file for Redis memory following the example: https://github.com/microsoft/autogen/blob/main/.github/workflows/pytest-mem0.yml.